### PR TITLE
Add C++ code

### PR DESCRIPTION
**Pull request by mplooster**
Added C++ wrapper code to make it a little easier (and hopefully intuitive) to work with the generated grpc code in C++. This PR does not include any generated code because it uses CMake to handle downloading gRPC and automatically generating the grpc code. For those that really don't want to use CMake, it also has instructions on how to generate the files using protoc.

### CODE REVIEW

File: grpc/cpp-cubic/README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Unlike other languages, C++ has more strict requirements to
+integrate with gRPC. The easiest way to accomodate these
+requirements is to build gRPC as part of the C++ project
+and use it to generate the required C++ files from the 
+cubic.proto file.
+
+## CMake Build
+To help simplify the build process, this project uses
+of [CMake](www.cmake.org). CMake will automatically download
+the grpc source code and include it as a subproject, giving
```
Comment (rslish) [inline grpc/cpp-cubic/README.md]: The capitalization of gRPC should be consistent through the document.

File: grpc/cpp-cubic/CobaltGRPC.cmake [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+# This allows cross-compiling to work - a native version of protoc is required.
+set(GRPC_NATIVE_DIR "" CACHE PATH "Path to native gRPC installation for cross compiling.")
+
+# Check if we have already added gRPC to the project. If not, we will
+# add it here.
+if(NOT TARGET grpc)
+    if(NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/grpc)
+        # Download grpc. We could do this as a git submodule instead, but
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/CobaltGRPC.cmake]: 👍 I agree with the decision not to make it a submodule.

File: grpc/cpp-cubic/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+and use it to generate the required C++ files from the 
+cubic.proto file.
+
+## CMake Build
+To help simplify the build process, this project uses
+[CMake](www.cmake.org). CMake will automatically download
+the grpc source code and include it as a subproject, giving
+access to the grpc libraries and the protoc compiler.
+
+To build as a standalone library execute the following
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/README.md]:
```suggestion
To build as a stand-alone library, execute the following
```

File: grpc/cpp-cubic/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+# can exist outside of the source code directory.
+mkdir build-cubic-client && cd build-cubic-client
+
+# Run CMake to download grpc and generate makefiles.
+# The final path specifies the directory that contains the
+# CMakeLists.txt file for the cubic_client C++ project.
+cmake -DCMAKE_BUILD_TYPE=Release <path/to/sdk-cubic/grpc/cpp-cubic>
+make cubic_client
+
+# To make a shared library, add -DBUILD_SHARED_LIBS=TRUE when running cmake
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/README.md]: We might want to clarify that this is a separate option, not that they'd run the steps twice.  Maybe something like
```
# Run CMake to download grpc and generate makefiles.
# To make a shared library, add -DBUILD_SHARED_LIBS=TRUE
# The final argument specifies the directory that contains the
# CMakeLists.txt file for the cubic_client C++ project.
cmake -DCMAKE_BUILD_TYPE=Release <path/to/sdk-cubic/grpc/cpp-cubic>
# OR
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=TRUE <path/to/sdk-cubic/grpc/cpp-cubic>

make cubic_client
```

File: grpc/cpp-cubic/cubic_client.h [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+private:
+    std::string mCubicVersion;
+    std::string mServerVersion;
+    std::vector<CubicModel> mModels;
+    unsigned int mTimeout;
+    std::unique_ptr<cobaltspeech::cubic::Cubic::Stub> mStub;
+
+    // Disable copy construction and assignments. Given the nature
+    // of the client, it's a bad idea to try to copy an existing connection.
```
Comment (renaissanceGeek) [inline grpc/cpp-cubic/cubic_client.h]: 👍 Good thinking!

**Comment by renaissanceGeek** [comment]
> I think this looks great! I couldn't comprehend the CMake stuff, so I'll leave it to someone else to approve.

@rslish It requires two approvals, and no one but Michael is a cmake expert yet :-)

**Comment by mplooster** [comment]
> I think it still needs the Documentation added to the webpage documentation

You are correct. I am hoping to do the documentation (and add example code) in another PR. For now, I've created issue #32 so we don't forget it.

**Pull request merged**